src/rewrapper.rs

```diff
@@ -119,6 +120,27 @@ fn exempt_from_wrapping(line: &str) -> bool {
     FULL_DT_TAG.is_match(line)
 }
 
+// Checking for if 'get diff' describes an addtion to 
+// already formatted lines, reformatting until we reach
+// the next empty newline
+fn mark_diff_lines_for_formatting(lines: &mut Vec<Line>) {
+    let mut should_format = false;
+
+    for i in 0..lines.len() {
+        if lines[i].should_format {
+            should_format = true;
+        }
+        if should_format {
+            // Condition to stop formatting at the next empty newline
+            if lines[i].contents.trim().is_empty() {
```

Comment:
Is this a strict enough condition? For example, what about:

```
<p>This is my paragraph
and it is great</p>
<p>This is my other paragraph
and it is also great</p>
```

If an addition is made in the middle of that somewhere, we want to make sure we don't mark the `<p>This is my other paragraph` like and all following ones as needed formatted, right? Specifically, my concern is that we might accidentally unwrap that second `<p>` onto the same line that the previous `</p>` is on. Does this PR do that, or is that avoided somehow?

(A test would probably be great for this).

Reply:
What do you think could be a possible solution to this? If we dont want to format on new open tags do you think formatting until next close tag would possibly work? I could do something like  if lines[i].contents.trim().is_empty() || lines[i].contains(OPEN_TAG) and we stop formatting there

Reply:
Right, but we can't just do any open tag, right? Because then we'd consider a bunch of things like `<span>` and `<var>` would be bad. What I'm saying is that there are so many kinds of open tags that shouldn't stop us from.

```
<p>This is my paragraph
and it is great. HOWEVER, ALL OF THIS CAPITALIZED TEXT WAS JUST <em>ADDED</em> IN
THE CURRENT BRANCH, AND IT VIOLATES THE FORMATTING RULES. And now we're back to
old text that existed before
the incorrectly-formatted
additions.</p>
```

For example, we can't.

So I have two suggestions:

1. Please add a test for the case I enumerate immediately above there, just to confirm that we never regress failing to wrap newly added lines just because they have a new
2. Can you confirm whether or not the example I describe in https://github.com/domfarolino/specfmt/pull/17#discussion_r1803186104 would break given the PR as it currently is? This might not be an issue after all, so I just want to confirm. So I think we should add a test for this scenario, and the one in (1) just above.

> What do you think could be a possible solution to this?

It'll be easier to know once I see the results of the tests I think we should add, but maybe this is as simple as just adding things like `</ol>` and `</p>` and some obvious other ones to https://github.com/domfarolino/specfmt/blob/master/src/rewrapper.rs#L104-L112?

Reply:
Commit a new with this testcase added, It does seem to wrap the second <p> onto the tag, an issue with adding </p> is this will only happen if the line ends in this close tag if we add text beyond the close tag on the same line as i did in this recent commit you can see it would completely ignore this check.

Reply:
A solution I see to this is formatting as we should until we encounter an open line tag like <p> or <ol> indicating a new entitiy which we would not want to format as it is seperate but this would then seem to ignore nested tags.

Reply:
I think the solution of formatting *until* we reach an open tag that cannot be nested is good. For example, we can start with the following list of open tags:
 - `<p>`
 - `<ol>`
 - `<ul>`
 - `<li>`

... and add more as we think of. It is not valid to nest those tags in HTML, so when we encounter them we can treat them as a symbol that forces us to push things to a new line. Does that sound OK to you?

Reply:
Sounds good i will give this a try